components/eamxx/src/physics/nudging/eamxx_nudging_process_interface.cpp

```diff
@@ -268,24 +342,39 @@ void Nudging::run_impl (const double dt)
     });
 
     // Vertical Interpolation onto atmosphere state pressure levels
+    // Note that we are going from ext to hxt here
     if (m_src_pres_type == TIME_DEPENDENT_3D_PROFILE) {
       perform_vertical_interpolation<Real,1,2>(p_mid_ext_p,
                                                p_mid_v,
                                                ext_state_view,
-                                               int_state_view,
+                                               hxt_state_view,
                                                int_mask_view,
                                                m_num_src_levs,
                                                m_num_levs);
     } else if (m_src_pres_type == STATIC_1D_VERTICAL_PROFILE) {
       perform_vertical_interpolation<Real,1,2>(p_mid_ext_1d,
                                                p_mid_v,
                                                ext_state_view,
-                                               int_state_view,
+                                               hxt_state_view,
                                                int_mask_view,
                                                m_num_src_levs,
                                                m_num_levs);
     }
 
+    // Refine-remap onto target atmosphere state horiz grid ("int")
+    // Note that we are going from hxt to int here
+    if(m_refine_remap) {
```

Comment:
Up to you, but if you want, you could do away with `m_refine_remap`, and simply do `if (refine_remapper)`, since `std::shared_ptr` has an `operator bool` overload...